python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -388,6 +399,43 @@ def terminate_scheduled_nodes(self):
         self.nodes_to_terminate = []
         self.update_worker_list()
 
+    def drain_nodes_via_gcs(self):
+        node_ips = [
+            self.provider.internal_ip(node_id)
+            for node_id in self.nodes_to_terminate
+        ]
+        raylet_ids_to_drain = {
+            self.load_metrics.raylet_id_by_ip[ip]
+            for ip in node_ips
+        }
+
+        logger.info(f"Draining raylets with ids {raylet_ids_to_drain}.")
+        request = gcs_service_pb2.DrainNodeRequest(drain_node_data=[
+            gcs_service_pb2.DrainNodeData(node_id=raylet_id)
+            for raylet_id in raylet_ids_to_drain
+        ])
+        try:
+            response = self.gcs_node_info_stub.DrainNode(request)
```

Comment:
Maybe comment the api guarantee here briefly?

Reply:
Good idea, will do that.